modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaJAXRSSpecServerCodegen.java

```diff
@@ -224,21 +224,26 @@ public void addOperationToGroup(String tag, String resourcePath, Operation opera
             basePath = basePath.substring(0, pos);
         }
 
-        String operationKey = basePath;
-        if (StringUtils.isEmpty(basePath)) {
-            basePath = tag;
-            operationKey = "";
-            primaryResourceName = tag;
-        } else if (basePath.matches("\\{.*\\}")) {
-            basePath = tag;
-            operationKey = "";
-            co.subresourceOperation = true;
-        } else {
-            if (co.path.startsWith("/" + basePath)) {
-                co.path = co.path.substring(("/" + basePath).length());
+        String operationKey = tag;
```

Comment:
I like this idea, however I think it should be guarded by an option the way other generators are doing this (see `useTags` in `SpringCodegen`. It looks like defaulting to tags will also have a negative impact on user upgrades for those relying on the subresource behaviors (or groupings in general). I think it would be best to have the opt-in option for 5.0 and make it default at some point later, maybe 5.1?

Reply:
like this idea. this should make this more a workaround rather than a breaking change. will change my approach